packages/flutter/lib/src/material/scaffold.dart

```diff
@@ -1023,15 +1023,17 @@ class ScaffoldState extends State<Scaffold> with TickerProviderStateMixin {
   bool _endDrawerOpened = false;
 
   void _drawerOpenedCallback(bool isOpened) {
-    setState(() {
-      _drawerOpened = isOpened;
-    });
+    _drawerOpened = isOpened;
+    if (hasEndDrawer) {
```

Comment:
Should we test this?

Reply:
Probably ... I will move to another PR (so I can confirm that it fixes the benchmarks too)